src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -727,6 +736,34 @@ public static Map<String, String> parseTopicToTableMap(String input) {
     return topic2Table;
   }
 
+  /**
+   * Convert a Comma separated key value pairs into a Map
+   *
+   * @param input Provided in KC config
+   * @return Map
+   */
+  public static Map<String, String> parseCommaSeparatedKeyValuePairs(String input) {
+    Map<String, String> pairs = new HashMap<>();
+    boolean isInvalid = false;
+    for (String str : input.split(",")) {
+      String[] tt = str.split(":");
+
+      if (tt.length != 2 || tt[0].trim().isEmpty() || tt[1].trim().isEmpty()) {
+        LOGGER.error(
+            "Invalid {} config format: {}",
+            SnowflakeSinkConnectorConfig.SNOWPIPE_STREAMING_CLIENT_PARAMETER_OVERRIDE_MAP,
+            input);
+        isInvalid = true;
+        break;
+      }
+      pairs.put(tt[0].trim(), tt[1].trim());
+    }
+    if (isInvalid) {
+      throw SnowflakeErrors.ERROR_0030.getException();
+    }
```

Comment:
Why not just throw above? Then you don't need to pass around `isInvalid`

Reply:
will do!